### PR TITLE
Updating aml validation to allow date of change edits and prevent form submission with no updates

### PR DESCRIPTION
**Pull request by cbroadley-ch**
Fixes for AML bug tickets:
[IDVA5-2107](https://companieshouse.atlassian.net/browse/IDVA5-2107)
[IDVA5-2135](https://companieshouse.atlassian.net/browse/IDVA5-2135)

[IDVA5-2107]: https://companieshouse.atlassian.net/browse/IDVA5-2107?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

- Exclude validation check to allow users to navigate to the 'When did AML supervision start?' date page only when editing/updating a new AML body that was added, this allows users to keep the existing aml number and make changes to the date entry.
- Validation captures duplicate aml body name/number if the user attempts to add new aml details for an aml that was previously added in ther journey
- Uses the existing AML_REMOVED_BODY_DETAILS session object to check for the edge case when a user removes aml details from the original acspFullProfile object and attempts to manually re-add the same details to acspUpdatedFullProfile - this now triggers the same error message as when duplicate aml entries are found (spoke with the PM and have agreed that this error message may change to better suit the edge case)
- Added unit tests 

[IDVA5-2135]: https://companieshouse.atlassian.net/browse/IDVA5-2135?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

**Review by ttingle-ch — APPROVED** [review]

File: src/controllers/features/update-acsp/updateYourDetailsController.ts [hunk 1]
```diff
@@ -52,10 +52,15 @@ export const get = async (req: Request, res: Response, next: NextFunction) => {
             correspondenceAddress: formatDateIntoReadableString(new Date(session.getExtraData(ACSP_UPDATE_CHANGE_DATE.CORRESPONDENCE_ADDRESS) || ""))
         };
 
-        // Format dateOfChange in removedAMLDetails
-        const removedAMLDetails: AmlSupervisoryBody[] = (session.getExtraData(AML_REMOVED_BODY_DETAILS) ?? []);
+        const removedAMLDetails: AmlSupervisoryBody[] = session.getExtraData(AML_REMOVED_BODY_DETAILS) || [];
+        const updatedRemovedAMLDetails = [...removedAMLDetails, ...acspFullProfile.amlDetails.filter((aml) => !acspUpdatedFullProfile.amlDetails.some(
```
Comment (rnesbitt-ch) [inline src/controllers/features/update-acsp/updateYourDetailsController.ts:56]: Just checking whether this is needed in this controller. Give it a test without and see

Comment (cbroadley-ch) [reply src/controllers/features/update-acsp/updateYourDetailsController.ts]: Thanks for the suggestion Ryan, as discussed I had made a copy of this session to ensure the original session object was kept immutable.
However, this is not required as we only perform checks for duplicates on AML_REMOVED_BODY_DETAILS within the amlMembershipNumberController and do not mutate the data that exists in this session.

I have now removed this redundant code.

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

**Review by rnesbitt-ch — APPROVED** [review]
